project_template/CMakeLists.txt

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)

```

Comment:
This may be too high of a minimum version to meet the required audience.

For example, the (only?) available ubuntu version in Azure DevOps Pipelines' Hosted Pool is 18.04 LTS. This version of ubuntu only has CMake 3.10 available in the core repositories[1].

The VS 2015 hosted image has CMake 3.7.1[2].
The VS 2017 hosted image has CMake 3.14.5[3].
The VS 2019 hosted image has CMake 3.14.5[4].

[1] https://packages.ubuntu.com/search?keywords=cmake&searchon=names&suite=bionic&section=all
[2] https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2015-Server2012R2-Readme.md
[3] https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2017-Server2016-Readme.md
[4] https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2019-Server2019-Readme.md

Reply:
We can install required version in AzureDevOps, of course if we need higher than 3.12. Ubuntu 16 now has 3.12.

Reply:
I think I'm OK with 3.14.5, but we want to decide if we want VERSION 3.14.5...3.15 or if we want to manually turn on CMP0091 when on 3.15 or above.

https://cmake.org/cmake/help/v3.15/policy/CMP0091.html#policy:CMP0091

Reply:
Also note: I was wrong about the version of ubuntu available -- looks like they only have 16.04 explicitly listed.